com.unity.netcode.gameobjects/TestHelpers/Runtime/NetcodeIntegrationTest.cs

```diff
@@ -111,6 +111,32 @@ public enum HostOrServer
 
         private NetworkManagerInstatiationMode m_NetworkManagerInstatiationMode;
 
+        private bool m_EnableVerboseDebug;
+
+        /// <summary>
+        /// Used to display the various integration test
+        /// stages and can be used to log verbose information
+        /// for troubleshooting an integration test.
+        /// </summary>
+        /// <param name="msg"></param>
+        protected void VerboseDebug(string msg)
+        {
+            if (m_EnableVerboseDebug)
+            {
+                Debug.Log(msg);
+            }
+        }
+
+        /// <summary>
+        /// Override this and return true if you need
+        /// to troubleshoot a hard to track bug within an
+        /// integration test.
+        /// </summary>
+        protected virtual bool OnSetVerboseDebug()
```

Comment:
just make the verbose bool flag protected instead?

Reply:
Yeah, I originally had that but forcing the override of that method will assure that the flag is set at the right time.
It basically is to avoid people setting it inside of their [UnityTest] attribute decorated method and wondering why it isn't providing all of the details.  
One example:
_During an integration test, a client throws an exception that has a stack trace that starts with the a client's message processing.  At first glance you might not realize that this was happening during the connection sequence if you set the flag within your [UnityTest] attribute decorated method because it wouldn't be enabled during the connection sequence._  

With it protected you could turn it on and off at different stages but could be missing important information that would help you figure out where in the "test runner stages" the issue was occurring.

With the virtual method you can't turn it on and off at different stages but you are assured to where in the "test runner stages" the exception is happening.  

There are pro's and con's to both approaches.  I took the virtual method path just to eliminate "another thing you have to remember all of the details" in order to yield the full benefits from it.